### DISCUSSION TITLE
Wildcards not reading properly?

### DISCUSSION THREAD

**Discussion by Vanir-Labs** [ImageMagick/ImageMagick#7014] [Help] [answered]
### ImageMagick version

7.1.1-26

### Operating system, version and so on

Windows 10

### Description

Hello,
I have several directories inside of several directories all with various filetypes. 
What I'm trying to do is select all, and only, the ***.bmp** files and run this convert script on them, _without changing their directories._
When I do the wildcards it gives me a file called **-1.file**. This is my script below...

```
magick "C:/Users/user/Desktop/fin/*.bmp" -format bmp -alpha off -type palette -colors 255 -depth 8 -compress none BMP3:"C:/Users/user/Desktop/fin/*.bmp"
pause
```
Please any help would be appreciated on this. I just want wildcard to select all .bmp files and process that line on each. I don't mind how long it takes.

**Comment by fmw42** [comment, marked as answer]
You cannot do both input and output wild cards in Imagemagick.  You can do input wildcards. If you want to process multiple images in a directory, then use magick mogrify rather than magick.  Alternately, create a script FOR loop to process each image one at a time in the loop.

**Comment by snibgo** [comment]
As fmw423 says.

Also, your command is simply `magick` (not `magick mogrify`) but have `-format bmp`, which is correct only for `magick mogrify`.

Note that `magick mogrify` uses one or more input filenames, often with a wildcard such as `*.bmp`. It doesn't take an output filename, because each output file is named after the corresponding input file.

You are reading all the BMP files in a directory, and writing BMP to the same directory, so the files will be overwritten.

**Comment by Vanir-Labs** [comment]
Thanks you two. I got it working with a for loop and mogrify.

**Reply by fmw42** [reply]
If you use a for loop, then use magick, not mogrify.  Mogrify is for processing multiple image (without a FOR loop). It can be used, but is not as functional as magick.  Magick is for processing a single image and should be used within a FOR loop.
